File: frontend_old/websites/src/website/packages/components/src/api/user/EditUsername.ts
import MakeRequest from "../MakeRequest";

export interface EditUsernameRequest {
  display_name: string;
}

export interface EditUsernameResponse {
  success: boolean;
  error_reason?: string;
}

export const EditUsername = MakeRequest<
  string,
  EditUsernameRequest,
  EditUsernameResponse,
  {}
>({
  method: "POST",
  routingFunction: () => "/v1/user/edit_username",
});

File: crates/api_clients/artcraft/artcraft_api_defs/src/users/edit_username.rs
use serde_derive::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Serialize, Deserialize, ToSchema)]
pub struct EditUsernameRequest {
  pub display_name: String,
}

#[derive(Serialize, Deserialize, ToSchema)]
pub struct EditUsernameResponse {
  pub success: bool,
}
